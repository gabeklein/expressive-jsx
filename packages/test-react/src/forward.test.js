const test = require("./_adapter");

describe("forward", () => {
  test("className", `
    () => {
      forward: className;

      <div>Hello World</div>
    }
  `);

  test("will join with existing styles", `
    () => {
      forward: className;
      color: red;

      <div>Hello World</div>
    }
  `);

  test("will pull className from existing props", `
    (props) => {
      forward: className;

      <div>Hello World</div>
    }
  `);

  test("will pull className from existing destructure", `
    ({ name }) => {
      forward: className;

      <div>Hello {name}</div>
    }
  `);

  test("will pull className from declared destructure", `
    (props) => {
      const { name } = props;

      forward: className;

      <div>Hello {name}</div>
    }
  `);

  test("will not break where ...rest exists", `
    ({ name, ...rest }) => {
      forward: className;

      <div {...rest}>
        Hello {name}
      </div>
    }
  `);

  // test("react ref", `
  //   ({ active }) => {
  //     input: {
  //       disabled = !active;
  //       color: red;
  //       forward: ref;
  //     }
      
  //     <this>
  //       <input type="text" />
  //     </this>
  //   }
  // `)
})