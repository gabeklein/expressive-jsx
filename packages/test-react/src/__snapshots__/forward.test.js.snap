// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forward className 1`] = `
"({
  className
}) => create(\\"div\\", {
  \\"className\\": className
}, create(\\"div\\", {}, \\"Hello World\\"));"
`;

exports[`forward className 2`] = `
"({
  className
}) => <div className={className}><div>Hello World</div></div>;"
`;

exports[`forward react ref 1`] = `
"forwardRef(({
  active
}, ref) => create(\\"input\\", {
  \\"disabled\\": !active,
  \\"ref\\": ref,
  \\"type\\": \\"text\\",
  \\"className\\": \\"input_16j\\"
}));
CSS.put(\`
	.input_16j { color: red }
\`);"
`;

exports[`forward react ref 2`] = `
"forwardRef(({
  active
}, ref) => <input disabled={!active} ref={ref} type=\\"text\\" className=\\"input_16j\\" />);
CSS.put(\`
	.input_16j { color: red }
\`);"
`;

exports[`forward will join with existing styles 1`] = `
"({
  className
}) => create(\\"div\\", {
  \\"className\\": use(\\"element_26b\\", className)
}, create(\\"div\\", {}, \\"Hello World\\"));

CSS.put(\`
	.element_26b { color: red }
\`);"
`;

exports[`forward will join with existing styles 2`] = `
"({
  className
}) => <div className={use(\\"element_26b\\", className)}><div>Hello World</div></div>;

CSS.put(\`
	.element_26b { color: red }
\`);"
`;

exports[`forward will not break where ...rest exists 1`] = `
"({
  className,
  name,
  ...rest
}) => create(\\"div\\", {
  \\"className\\": className
}, create(\\"div\\", Object.assign({}, rest), \\"Hello \\", name));"
`;

exports[`forward will not break where ...rest exists 2`] = `
"({
  className,
  name,
  ...rest
}) => <div className={className}><div {...rest}>Hello {name}</div></div>;"
`;

exports[`forward will pull className from declared destructure 1`] = `
"props => {
  const {
    className,
    name
  } = props;
  return create(\\"div\\", {
    \\"className\\": className
  }, create(\\"div\\", {}, \\"Hello \\", name));
};"
`;

exports[`forward will pull className from declared destructure 2`] = `
"props => {
  const {
    className,
    name
  } = props;
  return <div className={className}><div>Hello {name}</div></div>;
};"
`;

exports[`forward will pull className from existing destructure 1`] = `
"({
  className,
  name
}) => create(\\"div\\", {
  \\"className\\": className
}, create(\\"div\\", {}, \\"Hello \\", name));"
`;

exports[`forward will pull className from existing destructure 2`] = `
"({
  className,
  name
}) => <div className={className}><div>Hello {name}</div></div>;"
`;

exports[`forward will pull className from existing props 1`] = `
"props => {
  const {
    className
  } = props;
  return create(\\"div\\", {
    \\"className\\": className
  }, create(\\"div\\", {}, \\"Hello World\\"));
};"
`;

exports[`forward will pull className from existing props 2`] = `
"props => {
  const {
    className
  } = props;
  return <div className={className}><div>Hello World</div></div>;
};"
`;
